pkg/readers/adapters/amqp.go

```diff
@@ -59,3 +68,221 @@ func (a *Amqp) HandleAck(ackChannel <-chan messages.Ack) {
+	if internalExchange == "true" {
+		isInternalExchange = true
+	}
+
+	err = ch.ExchangeDeclare(
+		config.ExchangeName,
+		config.ExchangeType,
+		true,
+		false,
+		isInternalExchange,
+		false,
+		nil,
+	)
+	failOnError(err, "Failed to declare exchange")
+
+	if config.OuterExchangeName != "" {
+		err = ch.ExchangeDeclare(
+			config.OuterExchangeName,
+			config.OuterExchangeType,
+			true,
+			false,
+			false,
+			false,
+			nil,
+		)
+		failOnError(err, "Failed to declare outer exchange")
+		err = ch.ExchangeBind(
+			config.ExchangeName,
+			config.OuterExchangeBindingKey,
+			config.OuterExchangeName,
+			false,
+			nil,
+		)
+		failOnError(err, "Failed to bind outer exchange")
+	}
+	q, err := ch.QueueDeclare(
+		config.QueueName,
+		false,
+		false,
+		false,
+		false,
+		amqp.Table{"x-dead-letter-exchange": config.DeadLettersExchangeName},
+	)
+	failOnError(err, "Failed to declare queue")
+
+	err = ch.QueueBind(
+		q.Name,
+		config.QueueBindingKey,
+		config.ExchangeName,
+		false,
+		nil,
+	)
+	failOnError(err, "Failed to declare binding")
+
+	err = ch.Qos(config.QosPrefetchCount, 0, false)
+	failOnError(err, "Failed setting Qos")
+
+	msgs, err := ch.Consume(
+		q.Name,
+		config.ConsumerName,
+		false,
+		false,
+		false,
+		false,
+		nil,
+	)
+	if err != nil {
+		klog.Errorf("Failed to consume: %s", err)
+	}
+
+	return &Amqp{
+		ConsumedMessages: msgs,
+		Conn:             conn,
+		Ch:               ch,
+		Notify:           notify,
+	}, nil
+}
+
+func NewAmqpConfig() (amqpAdapter.Config, error) {
+	// @TODO this needs to be extracted to its own method
+	consumerId := os.Getenv("CONSUMER_ID")
+	if consumerId == "" {
+		// Work out consumer ID based on hostname: useful for k8s resources (pods controlled by deployment, statefulset)
+		hostname, err := os.Hostname()
+		if err != nil {
+			klog.Errorf("Could not set ConsumerID: %v", err)
+		}
+		pos := strings.LastIndex(hostname, "-")
+		consumerId = hostname[pos+1:]
+	}
+
+	// @TODO This needs to be extracted to its own object
+	data := struct {
+		ConsumerId string
+	}{
+		consumerId,
+	}
+
+	tpl := template.New("queueName")
+	tpl, err := tpl.Parse(helpers.GetEnv("RABBITMQ_QUEUE_NAME", ""))
+	if err != nil {
+		klog.Errorf("Invalid RABBITMQ_QUEUE_NAME: %v", err)
+	}
+	var buf bytes.Buffer
+	err = tpl.Execute(&buf, data)
+	if err != nil {
+		klog.Errorf("Invalid RABBITMQ_QUEUE_NAME: %v", err)
+	}
+	queueName := buf.String()
+
+	qosPrefetchCount, err := strconv.Atoi(os.Getenv("RABBITMQ_QOS_PREFETCH_COUNT"))
+	if err != nil {
+		qosPrefetchCount = 0
+	}
+
+	consumerName := os.Getenv("RABBITMQ_CONSUMER_NAME")
+	if consumerName == "" {
+		consumerName, _ = os.Hostname()
+	}
+
+	return amqpAdapter.Config{
+		Url:                     helpers.GetEnv("RABBITMQ_URL", ""),
+		DeadLettersExchangeName: helpers.GetEnv("RABBITMQ_DLX_NAME", ""),
+		DeadLettersQueueName:    helpers.GetEnv("RABBITMQ_DLX_QUEUE_NAME", ""),
+		ExchangeName:            helpers.GetEnv("RABBITMQ_EXCHANGE_NAME", ""),
```

Comment:
Is `ExchangeName` related to the inner exchange name? If so, maybe an improvement would be to add the inner prefix. WDYT?

Reply:
Yes, it is related to the inner exchange. I think there is no need to prefix this one since the outer exchange is already prefixed. But if we finally do it, we should change it also in the chart: https://github.com/softonic/homing-pigeon-chart